File: AtC/ABC/105/D.cpp
#include <bits/stdc++.h>
using namespace std;
#define ll long long

ll mod = 1e9+7;

int main() {
	ll n,m; cin>>n>>m;
	vector<ll> a(n);
	map<ll,ll> rem;

	rem[0]=1;
	ll sum=0, ans=0;
	for (ll i=0; i<n; i++) {
		cin>>a[i];

		sum = sum+a[i];
		rem[sum%m]++;

		ans += rem[sum%m]-1;
		ans %= mod;
	}
	
	cout << ans << endl;
	return 0;
}

File: AtC/ABC/105/D.py
n,m = [int(x) for x in input().split()]
a = [int(x) for x in input().split()]
rem = {}
rem[0]=1
sum=0
ans=0
for i in a:
    sum+=i
    if (sum%m in rem): rem[sum%m] += 1
    else: rem[sum%m] = 1
    ans += rem[sum%m]-1
print(ans)
